### PR TITLE
Check for newer classes in classpath for setting view owners

### PR DESCRIPTION
**Pull request by saket**

### CODE REVIEW

File: paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt [hunk 1]
```diff
@@ -633,10 +633,10 @@ class Paparazzi @JvmOverloads constructor(
       "androidx.lifecycle.ViewTreeLifecycleOwner"
     )
     private val hasSavedStateRegistryOwnerRuntime = isPresentInClasspath(
-      "androidx.savedstate.SavedStateRegistryOwner"
+      "androidx.savedstate.ViewTreeSavedStateRegistryOwner"
```
Comment (saket) [inline paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt:636]: `View#setViewTreeSavedStateRegistryOwner(owner)` is contained in `ViewTreeSavedStateRegistryOwner` and was introduced in [androidx.savedstate.*:1.2.0](https://developer.android.com/jetpack/androidx/releases/savedstate#1.2.0). 

Should we instead support a slightly older version of `1.1.0` by using the old `ViewTreeSavedStateRegistryOwner.set(view, owner)` API?

File: paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt [hunk 2]
```diff
@@ -633,10 +633,10 @@ class Paparazzi @JvmOverloads constructor(
...
     )
     private val hasAndroidxActivityRuntime = isPresentInClasspath(
-      "androidx.activity.OnBackPressedDispatcherOwner"
+      "androidx.activity.ViewTreeOnBackPressedDispatcherOwner"
```
Comment (saket) [inline paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt:639]: `View#setViewTreeOnBackPressedDispatcherOwner(owner)` is contained in `ViewTreeOnBackPressedDispatcherOwner` and was introduced in [androidx.activity.*:1.5.0](https://developer.android.com/jetpack/androidx/releases/activity#1.5.0).

File: paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt (hunk 1 above)
Comment (jrodbx) [reply paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt:636]: in general, we lean towards only supporting later versions of dependencies.  would it be too aggressive to expect paparazzi users to be on 1.2.0+?

**Review by jrodbx — APPROVED** [review]

File: paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt (hunk 1 above)
Comment (saket) [reply paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt:636]: I… don't have a good answer. Will requiring 1.2.0+ and calling it out in release notes be sufficient?

FWIW `1.2.0` was released in June, 2022 whereas `1.1.0` was released in Feb, 2022.

Comment (jrodbx) [reply paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt:636]: > Will requiring 1.2.0+ and calling it out in release notes be sufficient?

I think so.  :shipit:

Comment (saket) [reply paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt:636]: 🚀

Comment (joshfriend) [reply paparazzi/paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt:636]: Seems fine to me, this version is already bumping the minimum kotlin version and requiring AGP 7.4.0. Can't wait forever for consumers to catch up and definitely don't want to hold anyone back from using savedstate 1.2.0 either.

**Pull request merged**